test/translator/inquiry_test.cc

```diff
@@ -330,4 +325,135 @@ TEST_F(InquiryTest, TranslateUnitSerialNumberVpdNone) {
   char formatted_hex_string[31] = "12345abcde12345abcde_aaaaaaaa.";
   EXPECT_STREQ(product_serial_number, formatted_hex_string);
 }
-} // namespace
\ No newline at end of file
+
+TEST_F(InquiryTest, LogicalBlockProvisioningVpd) {
+  inquiry_cmd_ = scsi::InquiryCommand{
+      .evpd = 1, .page_code = scsi::PageCode::kLogicalBlockProvisioningVpd};
+
+  translator::StatusCode status =
+      translator::InquiryToScsi(scsi_cmd_, buffer, nvme_cmds_);
+  EXPECT_EQ(status, translator::StatusCode::kSuccess);
+
+  scsi::LogicalBlockProvisioningVpd result =
+      *reinterpret_cast<scsi::LogicalBlockProvisioningVpd*>(buffer);
+
+  bool ad = identify_ctrl.oncs.dsm;
+
+  EXPECT_EQ(result.page_code, scsi::PageCode::kLogicalBlockProvisioningVpd);
+  EXPECT_EQ(result.lbprz, ad);
+  EXPECT_EQ(result.anc_sup, identify_ns.nsfeat.thin_prov);
+  if (!ad && !identify_ns.nsfeat.thin_prov) {
+    EXPECT_EQ(result.provisioning_type, 0);
+  } else if (ad && !identify_ns.nsfeat.thin_prov) {
+    EXPECT_EQ(result.provisioning_type, 1);
+  } else if (ad && identify_ns.nsfeat.thin_prov) {
+    EXPECT_EQ(result.provisioning_type, 2);
+  }
```

Comment:
I'm not sure how I feel about having these if statements in your tests. Replicating the implementation logic in a test to test the implementation can be dangerous for two reasons:
1. if the implementation has a bug, the test will have the same bug and would fail to catch it.
2. Even if the specific replicated piece of logic is correct, when we test results conditionally like here, we might fail to notice if at some point the test starts taking path B instead of A (which might be undesirable).

Since your tests all define a specific `thin_prov ` and `ad` value combination, we can also have the test check only that specific path, instead of adding if clauses that will never be traversed by a given test. In this test for example it seems that both `add` and `thin_prov ` will always be `0`?